### ISSUE TITLE
OKEx Futures (OHLCV Historical Data): empty list returned by fetch_ohlcv

### ISSUE THREAD

**Issue by SergeGardien** [ccxt/ccxt#2900]
ATTENTION!!!

MUST READ THIS BEFORE SUBMITTING ISSUES:

https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#how-to-submit-an-issue

- OS: Mac OS X 10.11.6
- Programming Language version: Python 2.7
- CCXT version: 1.14.27
- Exchange: OKEx
- Method: fetch_ohlcv

Whenever I request OHLCV data for OKEx futures I get an empty list. Am I doing something wrong?

```python
exchange = ccxt.okex()

# These work
exchange.fetchTicker('BTC/USD')
exchange.fetchTicker('BTC/USD', {'contract_type': 'this_week'})
exchange.fetchTicker('BTC/USD', {'contract_type': 'next_week'})
exchange.fetchTicker('BTC/USD', {'contract_type': 'quarter'})

# This works
exchange.fetch_ohlcv('BTC/USD', "1m")

# These don't work, they return an empty list
exchange.fetch_ohlcv('BTC/USD', "1m", {'contract_type': 'this_week'})
exchange.fetch_ohlcv('BTC/USD', "1m", {'contract_type': 'next_week'})
exchange.fetch_ohlcv('BTC/USD', "1m", {'contract_type': 'quarter'})

**Comment by kroitor** [comment]
> # These don't work, they return an empty list
> exchange.fetch_ohlcv('BTC/USD', "1m", {'contract_type': 'this_week'})

The proper signature of `fetch_ohlcv` is:

```
fetch_ohlcv(symbol, timeframe, since, limit, params)
```

You're passing `{'contract_type': 'this_week'}` in the `since` argument, which is a mistake.

Documented here: https://github.com/ccxt/ccxt/wiki/Manual#ohlcv-candlestick-charts

I'd really recommend reading the entire Manual from the beginning to the end in order to understand this library and work with it effectively. Let me know if this does not answer your question.

**Comment by kroitor** [comment]
P.S. The proper code should be:

```
exchange.fetch_ohlcv('BTC/USD', "1m", None, None, {'contract_type': 'this_week'})
```

**Comment by kroitor** [comment]
> Next step is obviously to read the documentation.

Actually, that should be your first step when you approach any code ) It's your time and money, though, so... up to you )

**Comment by SergeGardien** [comment]
> that should be your first step when you approach any code

It depends on the work I have to do.
In this case I was looking to download historical data from different bitcoin exchanges. I've read the documentation of the websocket API of a specific exchange (I was planning to use it for the long-term) and when I started looking for some python examples your ccxt API popped up.
Hence I decided to give it a quick shot by coding some stuffs and getting what I needed.
I've searched also for other API wrappers for multiple bitcoin exchanges but your ccxt seems coming on top hence I've decided to settle for it.

Next step will be to read the documentation ;-)
